### ISSUE TITLE
Finish migration of Layer Presenter logic

### ISSUE THREAD

**Issue by Kartones** [CartoDB/cartodb#3674]
- [x] `fix app/controllers/carto/api/layers_controller.rb` `load_parent` method loading a `@parent` that can be either a user or a map
- [x] replace `Layer::Presenter` for `LayerPresenter` at all carto/xxxx models/controllers
- [x] Inside `LayerPresenter`, change either all fields to use `@xxxx` or setup `attr_reader`, but no current mixture
- [x] After #3670 goes live, need to recheck all `LayerPresenter` calls
- [x] <del>replace `options[:viewer_user]` inside `layer_presenter.rb` for `current_viewer `</del>
- [x] Clean and refactor code inside `LayerPresenter`
- [x] Add specs for both presenters <del>and migrate `spec/models/layer/presenter_spec.rb`</del>

Nice to have:
- Check usages of 'user_name', now should rely on presenter options `:viewer_user` and `:user` -> [Moved to separate ticket](https://github.com/CartoDB/cartodb/issues/3745)

Branch: `3674-layer-presenter-migration`
PR: #3702

**Comment by Kartones** [comment]
@juanignaciosl CR please. 

The `it_behaves_like` parameters might not be an awesome solution but does the job to reuse the same specs to run with different Presenters and Models so... at least works for now, can be improved in the future when we implement other CRUD operations in the refactored models.

I've backported some small bugfixes and tiny changes to add compatibility of `:viewer_user` etc. to old Presenter, but in general the sooner we switch to `Carto::LayerPresenter` the better.
